### ISSUE TITLE
Multiple output formats?

### ISSUE THREAD

**Issue by jrnail23** [stylelint/stylelint#4310]
> What is the problem you're trying to solve?

When running stylelint in our CI builds (we use CircleCI), I'd like to be able to generate a json file to be processed elsewhere (we upload it to SonarCloud), but I'd also like to see readable output in CircleCI so devs can easily see failure messages.

> What solution would you like to see?

I'd like to be able to specify multiple output formats and targets (i.e., string format for console, json format for output file)

For reference, I found this project that does the same thing for eslint: https://github.com/halkeye/eslint-formatter-multiple.

**Comment by vankop** [comment]
Fast solution will be write your custom formatter that runs all formatters you wants.

**Comment by ntwb** [comment]
Or fork halkeye/eslint-formatter-multiple, for stylelint, looks relatively simple.

**Comment by jrnail23** [comment]
yeah, @ntwb, I just realized that it is probably very similar.  At first I thought that stylelint formatters, by basically being serializers, wouldn't be appropriate for this, but it probably works the same.

**Comment by alejandroclaro** [comment]
@jrnail23 I would like you to reconsider this feature. Projects like eslint-formatter-multiple and eslint-output are more like a workaround and do not play nicely with monorepos (a lot of duplicated configuration).

I would be cleaner if the stylelint-cli supports this. These are some advantages:

* No additional / different CLI
* Single configuration file.
* Configuration in a single place and the configuration could be shared beetween multiple projetc like presets.

**Comment by jrnail23** [comment]
reopened at @alejandroclaro's request

**Comment by johnemau** [comment]
FWIW here is an inline script to build a custom formatter that writes to stdout and saves a json file by invoking the existing json and string formatters. JSON results are store in `r.json`.
```
echo "module.exports=(...a) => {require('fs').writeFileSync('r.json',require('stylelint').formatters.json(...a),{encoding:'utf8'});return require('stylelint').formatters.string(...a)}" > f.js
```
Then invoke stylelint with `--custom-formatter f.js` and remember to clean up the temporary file (`f.js`) after your done.
